### PR TITLE
feat(tile): add new decorator prop

### LINKED ISSUE #18001: Decorator: Tile

### PR DESCRIPTION
**Pull request by ariellalgilmore**
Closes #18001 

Update Tile's to use decorator prop
#### Changelog

**New**

- decorator prop

**Changed**

- deprecated slug prop
- updated, stories, test, and component to have a wrapper around decorator components

**Removed**

- svg hollowSlugIcon to use the imported AILabel icon

#### Testing / Reviewing

- [ ] remove tooltip story

### CODE REVIEW

File: packages/react/src/components/Tile/Tile.tsx [hunk 1]
```diff
@@ -113,13 +130,22 @@ Tile.propTypes = {
...
+      'been deprecated in favor of the new `decorator` prop. It will be removed in the next major release.'
+  ),
 };
 
 export interface ClickableTileProps extends HTMLAttributes<HTMLAnchorElement> {
   children?: ReactNode;
   className?: string;
 
+  /**
+   * **Experimental**: Provide a `decorator` component or set the boolean to True for an AILabel icon to be rendered inside the `TextInput` component
```
Comment (alisonjoseph) [inline packages/react/src/components/Tile/Tile.tsx]:
```suggestion
   * **Experimental**: Provide a `decorator` component or set the boolean to True for an AILabel icon to be rendered inside the `ClickableTile` component
```

File: packages/react/src/components/Tile/Tile.tsx [hunk 2]
```diff
@@ -292,6 +309,11 @@ ClickableTile.propTypes = {
    */
   clicked: PropTypes.bool,
 
+  /**
+   * **Experimental**: Provide a `decorator` component or set the boolean to True for an AILabel icon to be rendered inside the `TextInput` component
```
Comment (alisonjoseph) [inline packages/react/src/components/Tile/Tile.tsx]:
```suggestion
   * **Experimental**: Provide a `decorator` component or set the boolean to True for an AILabel icon to be rendered inside the `ClickableTile` component
```

File: packages/styles/scss/components/tile/_tile.scss [hunk 3]
```diff
@@ -539,22 +606,27 @@ $-icon-container-size: calc(#{layout.density('padding-inline')} * 2 + 1rem);
   }
 
   // To Do: Replace svg overrides with actual AI icon
```
Comment (alisonjoseph) [inline packages/styles/scss/components/tile/_tile.scss]:
```suggestion
```
I think we can remove this todo

**Review by alisonjoseph — COMMENTED** [review]
Looks great! Just a couple tiny things

File: packages/styles/scss/components/tile/_tile.scss (hunk 3 above)
Comment (ariellalgilmore) [reply packages/styles/scss/components/tile/_tile.scss]: lol idk how i left that

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
I think we can update the `Tile.featureflag.stories.js` to use the decorator also.

**Review by alisonjoseph — APPROVED** [review]
🥳

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**